lib/compiler.js

```diff
@@ -589,7 +589,8 @@ function resolveMiddlewarePath(rootDir, middleware, config) {
 
   // Try to require the module and check if <module>.<fragment> is a valid
   // function
-  var m = require(sourceFile);
+  var exports = require(sourceFile);
+  var m = exports && exports.__esModule ? exports.default : exports;
```

Comment:
This logic is duplicated on too many places. Could you please extract a shared helper function? Something along the following lines:

```js
// lib/require.js
module.exports = function requireNodeOrEsModule(sourceFile) {
  var exports = require(sourceFile);
  return exports && exports.__esModule ? exports.default : exports;
}
```